tests/test_triples_factory.py

```diff
@@ -393,50 +393,25 @@ def test_inverse_triples(self):
     def test_metadata(self):
         """Test metadata passing for triples factories."""
         t = Nations().training
-        self.assertEqual(NATIONS_TRAIN_PATH, t.metadata["path"])
-        self.assertEqual(
-            (
-                f"TriplesFactory(num_entities=14, num_relations=55, num_triples=1592,"
-                f' inverse_triples=False, path="{NATIONS_TRAIN_PATH}")'
-            ),
-            repr(t),
-        )
+        self.assertEqual(t.metadata, dict(path=NATIONS_TRAIN_PATH))
 
         entities = ["poland", "ussr"]
         x = t.new_with_restriction(entities=entities)
         entities_ids = t.entities_to_ids(entities=entities)
-        self.assertEqual(NATIONS_TRAIN_PATH, x.metadata["path"])
-        self.assertEqual(
-            (
-                f"TriplesFactory(num_entities=14, num_relations=55, num_triples=37,"
-                f' inverse_triples=False, entity_restriction={repr(entities_ids)}, path="{NATIONS_TRAIN_PATH}")'
-            ),
-            repr(x),
-        )
+        self.assertEqual(x.metadata, dict(path=NATIONS_TRAIN_PATH, entity_restriction=entities_ids))
 
         relations = ["negativebehavior"]
         v = t.new_with_restriction(relations=relations)
         relations_ids = t.relations_to_ids(relations=relations)
-        self.assertEqual(NATIONS_TRAIN_PATH, x.metadata["path"])
-        self.assertEqual(
-            (
-                f"TriplesFactory(num_entities=14, num_relations=55, num_triples=29,"
-                f' inverse_triples=False, path="{NATIONS_TRAIN_PATH}", relation_restriction={repr(relations_ids)})'
-            ),
-            repr(v),
-        )
+        self.assertEqual(v.metadata, dict(path=NATIONS_TRAIN_PATH, relation_restriction=relations_ids))
 
         w = t.clone_and_exchange_triples(t.triples[0:5], keep_metadata=False)
         self.assertIsInstance(w, TriplesFactory)
-        self.assertNotIn("path", w.metadata)
-        self.assertEqual(
-            "TriplesFactory(num_entities=14, num_relations=55, num_triples=5, inverse_triples=False)",
-            repr(w),
-        )
+        self.assertEqual(w.metadata, dict())
 
         y, z = t.split()
-        self.assertEqual(NATIONS_TRAIN_PATH, y.metadata["path"])
-        self.assertEqual(NATIONS_TRAIN_PATH, z.metadata["path"])
+        self.assertEqual(y.metadata, dict(path=NATIONS_TRAIN_PATH))
+        self.assertEqual(z.metadata, dict(path=NATIONS_TRAIN_PATH))
```

Comment:
does this make sense?

Reply:
This is fine, it's slickly more specific than before since it makes sure there isn't anything else in the metadata dictionary

Reply:
I was mostly wondering whether it makes sense to keep the path in the tf after splitting, since the new tf does *not* contain the triples from this path, but rather a subset thereof.

Reply:
ideally we could keep track of the transformations applied that gets to this point, but making  metadata model for this is hard. maybe instead, we could rename this `raw_path`?